### PR TITLE
Add animation settings

### PR DESCRIPTION
**Pull request by MrMeison**
Fix performance issues

### CODE REVIEW

File: src/visual.ts [hunk 1]
```diff
@@ -240,26 +240,41 @@ module powerbi.extensibility.visual {
         }
 
         private init(options: VisualConstructorOptions): void {
-            this.root = d3.select(options.element);
-
+            const root = d3.select(options.element);
```
Comment (ignatvilesov) [inline src/visual.ts]: Please specify a type of the variable.

File: src/visual.ts [hunk 2]
```diff
@@ -240,26 +240,41 @@ module powerbi.extensibility.visual {
...
+                    this.forceLayout.tick();
+                    this.forceLayout.resume();
                     d.isDrag = false;
                     this.fadeNode(d);
                 }))
-                .on('drag', ((d: ForceGraphNode) => this.fadeNode(d)));
-
-            this.colorPalette = options.host.colorPalette;
+                .on('drag', (d: ForceGraphNode) => {
+                    d.px += (d3.event as any).dx;
```
Comment (ignatvilesov) [inline src/visual.ts]: Why do you convert it to any? Can we use MouseEvent?

File: src/visual.ts [hunk 3]
```diff
@@ -240,26 +240,41 @@ module powerbi.extensibility.visual {
...
+                    d.x += (d3.event as any).dx;
+                    d.y += (d3.event as any).dy;
+                    this.fadeNode(d);
+                    this.forceLayout.tick();
+                });
 
-            this.tooltipServiceWrapper = createTooltipServiceWrapper(
-                options.host.tooltipService,
-                options.element);
+            const svg = root
```
Comment (ignatvilesov) [inline src/visual.ts]: Please specify a type of the variable.

File: src/visual.ts [hunk 4]
```diff
@@ -240,26 +240,41 @@ module powerbi.extensibility.visual {
...
+                .append('svg')
+                .attr({
+                    width: '100%',
```
Comment (ignatvilesov) [inline src/visual.ts]: Can we move it to CSS?

File: src/visual.ts [hunk 5]
```diff
@@ -682,47 +718,46 @@ module powerbi.extensibility.visual {
...
             // 'width/height * 20' seems enough to move nodes freely by force layout.
-            let maxWidth: number = viewport.width * resolutionFactor,
+            const maxWidth: number = viewport.width * resolutionFactor,
                 maxHeight: number = viewport.height * resolutionFactor,
-                limitX = x => Math.max((viewport.width - maxWidth) / 2, Math.min((viewport.width + maxWidth) / 2, x)),
-                limitY = y => Math.max((viewport.height - maxHeight) / 2, Math.min((viewport.height + maxHeight) / 2, y));
+                viewPortWidthDownLimit = (viewport.width - maxWidth) / 2,
+                viewPortHeightDownLimit = (viewport.height - maxHeight) / 2,
+                viewPortHeightUpLimit = (viewport.height + maxHeight) / 2,
+                viewPortWidthUpLimit = (viewport.height + maxHeight) / 2,
```
Comment (ignatvilesov) [inline src/visual.ts]: Please specify a type for each of these variables.

File: test/visualTest.ts [hunk 6]
```diff
@@ -231,6 +231,22 @@ module powerbi.extensibility.visual.test {
                             expect(element.css("font-size")).toBe(expectedFontSize);
                         });
                 });
+
+                it("animation turn off", (done) => {
```
Comment (ignatvilesov) [inline test/visualTest.ts]: I'd prefer `animation is turned off`.

File: test/visualTest.ts [hunk 7]
```diff
@@ -231,6 +231,22 @@ module powerbi.extensibility.visual.test {
...
+                    (dataView.metadata.objects as any).animation = { show: false };
+                    visualBuilder.updateRenderTimeout(dataView, () => {
+                        let originalTransform = visualBuilder.mainElement
```
Comment (ignatvilesov) [inline test/visualTest.ts]: Pleas specify a type.

File: test/visualTest.ts [hunk 8]
```diff
@@ -231,6 +231,22 @@ module powerbi.extensibility.visual.test {
...
+                            .children("g.node")
+                            .map((item, element) => element.getAttribute("transform"));
+                        setTimeout(() => {
+                            let afterFullRender = visualBuilder.mainElement
```
Comment (ignatvilesov) [inline test/visualTest.ts]: Pleas specify a type.

File: src/visual.ts (hunk 4 above)
Comment (MrMeison) [inline src/visual.ts]: Edge don't support `width` and `height` property as CSS rules

File: src/visual.ts (hunk 2 above)
Comment (MrMeison) [inline src/visual.ts]: No, `MouseEvent` doesn't have these properties

**Pull request merged**